gh-actions-vars.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+set -a
+
+# @environment ${ARCH}                    The architecture to build. Defaults to 'amd64'.
+# @environment ${DEBIAN_VERSION}          Debian version to build. Defaults to 'stretch'.
+# @environment ${DOCKER_HUB_REPO}         The docker hub repo to tag images for. Defaults to 'pihole'.
+# @environment ${DOCKER_HUB_IMAGE_NAME}   The name of the resulting image. Defaults to 'pihole'.
+
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD | sed "s/\//-/g")
+GIT_TAG=$(git describe --tags --exact-match 2> /dev/null || true)
+
+DEFAULT_DEBIAN_VERSION="stretch"
```

Comment:
we have a sprawling amount of default variables / updates to make in different places when those defaults change.  We don't need it immediately but at some point, I'd like to centralize them with a very early export of defaults in a centrally included envs.sh that the pipeline uses before any builds/tests/uploads run.  variables in the pipeline would work as a central location too but consider executing issues executing locally may have when putting things in the pipelines. The version number could migrate to this environment export too potentially.